core/gather/gatherers/dobetterweb/domstats.js

```diff
@@ -81,6 +81,43 @@ class DOMStats extends FRGatherer {
     supportedModes: ['snapshot', 'navigation'],
   };
 
+  /**
+   * @param {LH.Crdp.Page.FrameTree} rootFrameTree
+   * @return {Pick<LH.Artifacts.DOMStats, "totalFrames" | "framesDepth">}
+   */
+  static determineFrameStats(rootFrameTree) {
+    let maxDepth = 0;
+    let maxFrameUrl = '';
+    let totalFrames = 0;
+
+    /**
+     * @param {LH.Crdp.Page.FrameTree} frameTree
+     */
+    function walk(frameTree, depth = 1) {
+      // When Chrome comes across an iframe for a URL already framed above in the hierarchy,
+      // it will load as an empty document and its URL in the frame tree will be just `:`.
+      // Best to just ignore.
```

Comment:
What does "framed above in the hierarchy mean"?

Reply:
URL A -> URL B -> URL C -> URL B (oops, would recurse...but chrome steps in)